src/placeos-rest-api/controllers/drivers.cr

```diff
@@ -193,5 +196,13 @@ module PlaceOS::Api
       # Find will raise a 404 (not found) if there is an error
       Model::Driver.find!(id, runopts: {"read_mode" => "majority"})
     end
+
+    protected def can_read
+      can_scopes_read("drivers")
```

Comment:
Something that might reduce a bit of code duplication could be making the controller scope an abstract def

```crystal
abstract class Application < ...
  abstract def controller_scope_resource : String
  
  protected def can_write?
    can_scopes_write(controller_scope_resource)
  end
  
  protected def can_read?
    can_scopes_read(controller_scope_resource)
  end
end

...
  
class Drivers < Application
  before_action :can_read, only: [:index, :show]
  before_action :can_write, only: [:create, :update, :destroy, :remove, :update_alt]
    
  # Then you only need to define this in each controller before defining the filters
  getter controller_scope_resource : String = "drivers"
end
```

Reply:
This does look like a good addition

Reply:
Might turn this into a issue and do it next sprint